src/settings/SettingsStore.js

```diff
@@ -138,7 +141,7 @@ export default class SettingsStore {
             settingName = setting.invertedSettingName;
         }
 
-        const watcherId = `${new Date().getTime()}_${settingName}_${roomId}`;
+        const watcherId = `${SettingsStore._watcherCount++}_${settingName}_${roomId}`;
```

Comment:
I'd still keep the time just in case we do have a race or something. More randomness isn't a bad thing.